### PR TITLE
Added the documentation comments on the api/system, api/clipboard; co…

### PR DESCRIPTION
**Pull request by andreGarvin**
…uld not document two methods of the system class excuteOnRemote().

### CODE REVIEW

File: src/api/application/application.ts [hunk 1]
```diff
@@ -15,29 +15,39 @@ export class NavigationRejectedReply extends Reply<'window-navigation-rejected',
...
 */
 export class Application extends Base {
 
     /**
-        @param { object } wire
-        @param { object } identity
-        @constructor
+     * @param { object } wire
+     * @param { object } identity
+     * @constructor
```
Comment (rdepena) [inline src/api/application/application.ts]: @andreGarvin lets not document any of these constructors that receive a wire, these are internal use only.

File: src/api/application/application.ts [hunk 2]
```diff
@@ -15,29 +15,39 @@ export class NavigationRejectedReply extends Reply<'window-navigation-rejected',
     public url: string;
 }
 
+
 export default class ApplicationModule extends Bare {
+    /**
+     * Returns an Application object that represents an existing application.
+     * @param { object } indentity
```
Comment (rdepena) [inline src/api/application/application.ts]: Identity is a type, we could jsdoc typedef it.

File: src/api/application/application.ts [hunk 3]
```diff
@@ -74,28 +84,37 @@ export class Application extends Base {
...
+      * @param { boolean } force sets the value force to false
+      * @return {Promise.<boolean>}
     */
     public close(force: boolean = false): Promise<void> {
         return this.wire.sendAction('close-application', Object.assign({}, this.identity, {force})).then(() => undefined);
     }
 
+    /**
+     * Retrieves an array of wrapped fin.desktop.Windows for each of the application’s child windows.
+     * @return {Promise.<array>}
```
Comment (rdepena) [inline src/api/application/application.ts]: This is a typed array, we should refer to the _Window class here.

File: src/api/interappbus/interappbus.ts [hunk 4]
```diff
@@ -16,20 +16,20 @@ export default class InterApplicationBus extends Bare {
     private refCounter = new RefCounter();
 
     /**
-      @param { object } wire
-      @constructor
+     * @param { object } wire
+     * @constructor
     */
```
Comment (rdepena) [inline src/api/interappbus/interappbus.ts]: internal constructors do not need to be documented.

File: src/api/interappbus/interappbus.ts [hunk 5]
```diff
@@ -56,6 +56,16 @@ export default class InterApplicationBus extends Bare {
         }).then(() => undefined);
     }
 
+    /**
+     * Subscribes to messages from the specified application on the specified topic.
+     * If the subscription is for a uuid, [name], topic combination that has already
+     * been published to upon subscription you will receive the last 20 missed messages
+     * in the order they were published.
+     * @param source
```
Comment (rdepena) [inline src/api/interappbus/interappbus.ts]: these are missing the types.

File: src/api/interappbus/interappbus.ts [hunk 6]
```diff
@@ -76,6 +86,13 @@ export default class InterApplicationBus extends Bare {
         return this.refCounter.actOnFirst(subKey, sendSubscription, alreadySubscribed);
     }
 
+    /**
+     * Unsubscribes to messages from the specified application on the specified topic.
+     * @param source 
+     * @param topic
```
Comment (rdepena) [inline src/api/interappbus/interappbus.ts]: these are missing the types.

File: src/api/notification/notification.ts [hunk 7]
```diff
@@ -97,9 +97,9 @@ export class _Notification extends Base implements Notification {
     }
 
     /**
-      @param { object } options
-      @param { object } wire
-      @constructor
+     * @param { object } options
+     * @param { object } wire
+     * @constructor
```
Comment (rdepena) [inline src/api/notification/notification.ts]: private constructors do not need to be documented.

File: src/api/notification/notification.ts [hunk 8]
```diff
@@ -147,9 +146,9 @@ export class _Notification extends Base implements Notification {
     }
 
     /**
-      Sends a message to the notification.
-      @params {*} message
-      @return {Promise.<void>}
+     * Sends a message to the notification.
+     * @param {*} message
```
Comment (rdepena) [inline src/api/notification/notification.ts]: I have seen `any` used in other parts of the code. we should use one nomenclature.

File: src/api/window/window.ts [hunk 9]
```diff
@@ -21,17 +21,20 @@ export interface CloseEventShape {
...
 
     /**
-      @param { object } wire
-      @param { object } identity
-      @constructor
+     * @param { object } wire
+     * @param { object } identity
+     * @constructor
     */
     constructor(wire: Transport, public identity: Identity) {
```
Comment (rdepena) [inline src/api/window/window.ts]: private constructors do not need to be documented.

File: src/api/window/window.ts [hunk 10]
```diff
@@ -113,32 +135,63 @@ export class _Window extends Base {
...
+     */
     public stopFlashing(): Promise<void> {
         return this.wire.sendAction('stop-flash-window', this.identity).then(() => undefined);
     }
 
+    /**
+     * Retrieves an array containing wrapped fin.desktop.Windows that are grouped with this
+     * window. If a window is not in a group an empty array is returned. Please note that
+     * calling window is included in the result array.
+     * @return {Promise.<array>}
```
Comment (rdepena) [inline src/api/window/window.ts]: typed array should be used here.

File: src/api/application/application.ts (hunk 2 above)
Comment (andreGarvin) [inline src/api/application/application.ts]: okay

File: src/api/notification/notification.ts (hunk 8 above)
Comment (andreGarvin) [inline src/api/notification/notification.ts]: nomenclature ? @rdepena

File: src/api/window/window.ts (hunk 10 above)
Comment (andreGarvin) [inline src/api/window/window.ts]: What do you mean by typed array @rdepena ?

File: src/api/application/application.ts [hunk 11]
```diff
@@ -74,28 +83,37 @@ export class Application extends Base {
...
+      * @param { boolean } force sets the value force to false
+      * @return {Promise.<boolean>}
     */
     public close(force: boolean = false): Promise<void> {
         return this.wire.sendAction('close-application', Object.assign({}, this.identity, {force})).then(() => undefined);
     }
 
+    /**
+     * Retrieves an array of wrapped fin.desktop.Windows for each of the application’s child windows.
+     * @return {Promise.<_Window>}
```
Comment (rdepena) [inline src/api/application/application.ts]: Wait, these should be a typed Array.

Comment (rdepena) [inline src/api/application/application.ts]: https://stackoverflow.com/questions/14611995/how-to-specify-an-array-of-objects-as-a-parameter-or-return-value-in-jsdoc#14854851

**Pull request merged**